src/landscape.cpp

```diff
@@ -865,7 +865,8 @@ static void GenerateTerrain(int type, uint flag)
 	uint x = r & MapMaxX();
 	uint y = (r >> MapLogX()) & MapMaxY();
 
-	if (x < 2 || y < 2) return;
+	if (x < (uint)2 + (_settings_game.construction.freeform_edges ? 1 : 0)) return;
+	if (y < (uint)2 + (_settings_game.construction.freeform_edges ? 1 : 0)) return;
```

Comment:
`freeform_edges` doesn't exist for the original generator, so this is confusing (and wrong I think?)

Reply:
Owh, it does, so this is correct; but that still makes me unclear why it wouldn't be needed on the other side of the map ;)